yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -83,6 +84,9 @@ private[yarn] class YarnAllocator(
   private var executorIdCounter = 0
   @volatile private var numExecutorsFailed = 0
 
+  @volatile private var executorFailureTimeStamps = new Stack[Long]()
```

Comment:
Do we not want to use a queue here, because the first executor failures to leave the list should be the first ones that were added?